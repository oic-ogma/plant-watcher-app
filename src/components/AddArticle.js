import React, { Component } from 'react';
import { View } from 'react-native';
import { Label, FormInputComponent } from './common';
import { FormValidationMessage, Button } from 'react-native-elements';

export default class AddArticle extends Component {
  saveArticle() {
    const { plantName, articleContents, saveArticle } = this.props;
    saveArticle(plantName, articleContents);
  }

  render() {
    const {
      plantName,
      articleContents,
      plantNameChanged,
      articleContentsChanged,
      error
    } = this.props;

    return (
      <View>
        <Label placeholder='植物名' />
        <FormInputComponent
          onChangeText={plantNameChanged}
          placeholder='ひまわり'
          value={plantName}
        />

        <Label placeholder='記事'/>
        <FormInputComponent
          placeholder='黄色い'
          onChangeText={articleContentsChanged}
          multiline={true}
          value={articleContents}
        />
        <FormValidationMessage labelStyle={{ fontSize: 16 }}>{error}</FormValidationMessage>

        <Button
          onPress={this.saveArticle.bind(this)}
          title='投稿'
          raised
          large
          loading={this.props.loading}
          backgroundColor='green'
        />

      </View>
    );
  }
}
