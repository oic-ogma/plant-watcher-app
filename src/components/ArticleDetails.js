import React, { Component } from 'react';
import { View, Text, Image, StyleSheet, ScrollView } from 'react-native';
import { Rating } from 'react-native-elements';
import { Spinner } from './common';
import { Icon } from 'react-native-elements';

export default class ArticleDetails extends Component {
  componentWillMount() {
    const { fetchArticleDetails, fetchArticleDecision } = this.props;
    fetchArticleDetails();
    fetchArticleDecision();
  }

  computeDuration(ms) {
    const dt = new Date();
    const year = dt.getFullYear(ms);
    const month = dt.getMonth(ms);
    const date = dt.getDate(ms);

    return year + '/' + month + '/' + date;
  }

  render() {
    const {
      body,
      editOrBookmarkBlock,
      block1,
      uploadImage,
      blockChild1,
      articleTitle,
      uploadDay,
      block2,
      bookmarked,
      article,
      review,
      border,
      rating,
    } = styles;

    const {
      plantName,
      articleContents,
      createdAt,
      currentRating,
      articleDecision,
    } = this.props;

    const editOrBookmark = () => {
      if (articleDecision === true) {
        const style = {
          fontSize: 18,
          color: '#6BC6FD',
          paddingVertical: 5,
          paddingHorizontal: 10,
          borderStyle: 'solid',
          borderWidth: 1,
          borderColor: '#6BC6FD',
          borderRadius: 3
        };
        return (
          <Text style={style}>編集</Text>
        );
      } else {
        return (
          <Icon type='font-awesome' name='bookmark-o' color='#D04255' size={30}/>
        );
      }
    };

    if (currentRating) {
      return (
        <ScrollView style={body}>
          <View style={editOrBookmarkBlock}>
            {editOrBookmark()}
          </View>
          <View style={block1}>
            <Image
              source={{ url: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcT4pZ0SHqgZhy-_ATEzJPpqXL3241NN8_Y0TKMiNSdO0rZc6STa7A' }}
              style={uploadImage}
            />
            <View style={blockChild1}>
              <Text style={articleTitle}>{plantName}</Text>
              <Text style={uploadDay}>{this.computeDuration(createdAt)}</Text>
            </View>
          </View>
          <View style={block2}>
            <Text style={bookmarked}>0ブックマーク</Text>
            <Rating
              type='star'
              readonly
              imageSize={18}
              startingValue={currentRating}
            />
          </View>
          <View style={border}/>
          <View>
            <Text style={article}>{articleContents}</Text>
          </View>
          <Text style={review}>レビュー</Text>
          <View style={border}/>
          <Rating
            type='star'
            showRating
            startingValue={3}
            imageSize={30}
            onFinishRating={this.ratingCompleted}
            style={rating}
          />
        </ScrollView>

      );
    }
    return <Spinner/>;
  }
}

const styles = StyleSheet.create({
  body: {
    padding: 30,
  },
  editOrBookmarkBlock: {
    position: 'absolute',
    top: -10,
    right: 5
  },
  block1: {
    display: 'flex',
    flexDirection: 'row'
  },
  uploadImage: {
    width: 80,
    height: 80
  },
  blockChild1: {
    marginLeft: 18,
  },
  articleTitle: {
    paddingBottom: 5,
    fontSize: 21
  },
  uploadDay: {
    color: '#95989A',
    fontSize: 12
  },
  block2: {
    display: 'flex',
    alignItems: 'center',
    flexDirection: 'row',
    marginTop: 15,
    marginBottom: 20
  },
  bookmarked: {
    marginRight: 18,
    fontSize: 12
  },
  article: {
    fontSize: 15,
    color: '#555555',
    letterSpacing: 1.5,
    lineHeight: 15 * 1.3,
  },
  review: {
    textAlign: 'center',
    marginTop: 40,
    marginBottom: 5,
    fontSize: 18
  },
  border: {
    marginBottom: 12,
    borderBottomColor: '#D1D1D6',
    borderBottomWidth: 1
  },
  rating: {
    marginBottom: 60
  }
});