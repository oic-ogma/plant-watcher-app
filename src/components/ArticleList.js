import React, { Component } from 'react';
import { ListItem, Error, Spinner } from './common';
import { ListView } from 'react-native';



class ArticleList extends Component {


  componentWillMount() {
    this.props.getMyArticles();
    this.createDataSource(this.props);
  }

  componentWillReceiveProps(nextProps) {
    this.createDataSource(nextProps);
  }

  createDataSource({ articles }) {
    const ds = new ListView.DataSource({
      rowHasChanged: (r1, r2) => r1 !== r2
    });

    this.dataSource = ds.cloneWithRows(articles);
  }

  renderRow({ plantName, articleContents, error }) {
    return <ListItem plantName={plantName} articleContents={articleContents} error={error} />;
  }

  render() {
    if (this.props.loading) {
      return <Spinner size='large' />;
    }
    if (this.props.error) {
      return (<Error message={this.props.error} />);
    }
    return (
      <ListView
        enableEmptySections
        dataSource={this.dataSource}
        renderRow={this.renderRow}
      />
    );
  }
}

export default ArticleList;
