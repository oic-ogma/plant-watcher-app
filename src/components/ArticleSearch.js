import React, { Component } from 'react';
import { Button } from  'react-native-elements';
import { View, Keyboard } from 'react-native';
import { FormValidationMessage } from 'react-native-elements';
import { SearchBarComponent } from './common/SearchBar';
import { Actions } from 'react-native-router-flux';

export default class ArticleSearch extends Component {
  onSubmit() {
    const { textSearchArticle, plantName } = this.props;
    Keyboard.dismiss();
    textSearchArticle(plantName);
  }
  render() {
    const { searchArticlePlantNameChanged, plantName, error } = this.props;
    const { buttonStyle } = styles;

    return (
      <View style={{ marginTop: 140 }}>
        <SearchBarComponent
          onChangeText={searchArticlePlantNameChanged}
          placeholder='植物名'
          onSubmitEditing={this.onSubmit.bind(this)}
          value={plantName}
        />

        <FormValidationMessage>
          {error}
        </FormValidationMessage>

        <Button
          raised
          large
          icon={{ name: 'camera-alt' }}
          containerViewStyle={buttonStyle}
          title='カメラで撮影して検索'
          backgroundColor='green'
          onPress={Actions.imagesearch}
        />
      </View>
    );
  }
}

const styles = {
  buttonStyle: {
    marginTop: 50,
    marginBottom: 30
  }
};
