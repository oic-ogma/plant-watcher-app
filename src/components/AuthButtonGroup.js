import React from 'react';
import  { Actions } from 'react-native-router-flux';
import { Button } from 'react-native-elements';
import { View } from 'react-native';

export const AuthButtonGroup = () => {
  const { buttonStyle, button } = styles;
  return (
    <View style={buttonStyle}>
      <Button large buttonStyle={button} onPress={Actions.articleSearch} title='ログイン' />
      <Button onPress={Actions.articleSearch} title='ユーザー登録' />
    </View>
  );
};

const styles = {
  buttonStyle: {
    width: '100%',
    flex: 1,
    flexDirection: 'row',
    alignItems: 'flex-end',
    justifyContent: 'flex-start',
    paddingLeft: '20%'
  },
  button: {
    marginTop: 100,
  }
};
