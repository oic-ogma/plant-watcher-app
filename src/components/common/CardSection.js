import React from 'react';
import { StyleSheet, View } from 'react-native';

export const CardSection = props => (
  <View style={[styles.containerStyle, props.style]}>
    {props.children}
  </View>
);

const styles = StyleSheet.create({
  containerStyle: {
    borderBottomWidth: 1,
    padding: 5,
    backgroundColor: '#fff',
    justifyContent: 'flex-start',
    flexDirection: 'row',
    borderColor: '#ddd',
    position: 'relative',
  }
});
