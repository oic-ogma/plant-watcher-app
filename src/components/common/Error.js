import React from 'react';
import { Text } from 'react-native';

export const Error = ({ message }) => (
  <Text style={styles.errorTextStyle}>
    {message}
  </Text>
);

const styles = {
  errorTextStyle: {
    paddingTop: 5,
    fontSize: 18,
    alignSelf: 'center',
    color: '#FF0000',
  }
};
