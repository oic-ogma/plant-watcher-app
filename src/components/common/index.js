export * from './Button';
export * from './Card';
export * from './CardSection';
export * from './Header';
export * from './Input';
export * from './Spinner';
export * from './Confirm';
export * from './Error';
export * from './ListItem';
export * from './ListArticles';
export * from './Label';
export * from './FormInput';
export * from './IconWrapper';