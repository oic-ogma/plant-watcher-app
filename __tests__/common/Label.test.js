import React from 'react';
import { Label } from '../../src/components/common';
import renderer from 'react-test-renderer';

describe('LabelComponent', () => {
  it('snapshot-普通', () => {
    const tree = renderer.create(<Label/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('snapshot-placeholderが渡されている', () => {
    const tree = renderer.create(<Label placeholder='ひまわり' />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
