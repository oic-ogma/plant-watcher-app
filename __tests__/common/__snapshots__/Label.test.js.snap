// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelComponent snapshot-placeholderが渡されている 1`] = `
<View
  style={
    Array [
      Object {},
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {},
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        Array [
          Object {
            "color": "#86939e",
            "fontSize": 15,
            "fontWeight": "bold",
            "marginBottom": 1,
            "marginLeft": 20,
            "marginRight": 20,
            "marginTop": 15,
          },
          Object {
            "fontSize": 20,
          },
          undefined,
        ],
      ]
    }
  >
    ひまわり
  </Text>
</View>
`;

exports[`LabelComponent snapshot-普通 1`] = `
<View
  style={
    Array [
      Object {},
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {},
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        Array [
          Object {
            "color": "#86939e",
            "fontSize": 15,
            "fontWeight": "bold",
            "marginBottom": 1,
            "marginLeft": 20,
            "marginRight": 20,
            "marginTop": 15,
          },
          Object {
            "fontSize": 20,
          },
          undefined,
        ],
      ]
    }
  />
</View>
`;
